### PR TITLE
azcore cleanup

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Refactored various policy options to conform with our options pattern.
Removed some content that was specific to storage.
Made some content internal as it's not needed for consumers.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/azcore/policy_telemetry_test.go [hunk 1]
```diff
@@ -37,7 +37,9 @@ func TestPolicyTelemetryWithCustomInfo(t *testing.T) {
 	defer close()
 	srv.SetResponse()
 	const testValue = "azcore_test"
-	pl := NewPipeline(srv, NewTelemetryPolicy(TelemetryOptions{Value: testValue}))
+	o := DefaultTelemetryOptions()
+	o.Value = testValue
```
Comment (catalinaperalta) [inline sdk/azcore/policy_telemetry_test.go:41]: Are we checking that this is properly assigned in the test?

Comment (catalinaperalta) [reply sdk/azcore/policy_telemetry_test.go:41]: Well we cant really since it's part of telemetry..

Comment (jhendrixMSFT) [reply sdk/azcore/policy_telemetry_test.go:41]: We do check it below, it's just part of the `User-Agent` request header.

Comment (catalinaperalta) [reply sdk/azcore/policy_telemetry_test.go:41]: Ah yes, I saw it. Thanks for pointing it out.

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**